### DISCUSSION TITLE
🐶 Pet Adopt Soroban Dapp Tutorial 🐶: Learn Soroban The Fun Way

### DISCUSSION THREAD

**Discussion by esteblock** [stellar/sorobanathon#165] [General]
Hello!  I wanted to make Soroban development accessible to all kind of developers, for all levels. To that end, and based on the [Ethereum Pet Shop App ](https://trufflesuite.com/blog/learn-ethereum-the-fun-way-with-our-pet-shop-tutorial/), I developed the **Pet Adopt Soroban Dapp**.

![screenshot](https://user-images.githubusercontent.com/2274485/208001087-bd337740-435b-4175-bd77-d9fed1a553e1.png)

 It covers the entire local development process, from start to finish. 
 
 This project is divided in 3 parts:

1.  The [pet-adopt-soroban-contracts ](https://github.com/esteblock/pet-adopt-soroban-contracts) repo.
Repository with the necessary Smart Contracts to adopt pets, see who did adopted a certain pet, and check if a pet has been adopted or not. It does use the latest version of soroban-sdk.

2. The [pet-adopt-soroban ](https://github.com/esteblock/pet-adopt-soroban)front-end project.
A ReactJS & Next.js front end project that connects to your Freighter wallet, reads the blockchain and sign messages in order to adopt cute pets.

The project uses the @soroban-react packages! The front end has been coded from 0 using reactjs, however I took the pet images and names from the [truffle'spet-shop-box](https://github.com/truffle-box/pet-shop-box).

3. The [Pet Adopt Tutorial] (https://pet-adopt.gitbook.io/untitled/): A step-by-step tutorial that explains you how to build your first dapp!
It does start from a simple, but already working reactJS webpage, but without any interaction/package related to soroban  or any wallet. And in 5 chapters it shows how to build your first dapp (create & deploy contracts, add wallet interaction, read from soroban, sign messages & write the blockchain)

Check it in: https://pet-adopt.gitbook.io/untitled/

As a disclaimer: I must say that due to the recent changes in the soroban-client, xdr versions and more, the app has  some bugs. But is open for collaborations!!!
And also it took me muuch more time than expected (specially dealing with the @soroban-react libraries). However is my objetive to continue supporting this tutorial/project with the future versions of soroban-sdk, xdr... I also want to expand the tutorial  making it more clear, with more details of every step, with screenshots, etc...

You can follow my discussions in: [discord](https://discord.com/channels/897514728459468821/1045347966476628108/1052639383628562543) or the [issue](https://github.com/stellar/soroban-example-dapp/issues/72) I have open

**Comment by esteblock** [comment]
Also, the tutorial is also hosted in github! 
So anyone can collaborate! https://github.com/esteblock/pet-adopt-soroban/tree/main/docs

**Reply by esteblock** [reply]
Edited some typos

**Comment by achallagundla** [comment]
Woah this is cool! 
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission may qualify for a reward of 1500 XLM! One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately.

Please let us know if you have any questions. (You can also message me on discord @ anuxhya)

All the best,
Anuhya

PS: No need to submit the form if you already have, just update me on discord if your wallet address has changed.

**Reply by esteblock** [reply]
Sent you DM on Discord
